packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -83,142 +88,76 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
 
 /// TODO: DOC add docstrings on impl'd methods
 impl<'s> JsPackage<'s> {
-    fn import(
-        scope: &mut v8::HandleScope<'s>,
-        context: v8::Local<'s, v8::Context>,
-        embedded: &Embedded<'s>,
-        name: &str,
-        pkg_type: PackageType,
-    ) -> Result<Self> {
+    fn import(scope: &mut v8::HandleScope<'s>, name: &str, pkg_type: PackageType) -> Result<Self> {
         let path = get_pkg_path(name, pkg_type);
         tracing::debug!("Importing package from path `{path}`");
-        let code = match fs::read_to_string(path.clone()) {
-            Ok(s) => s,
-            Err(_) => {
-                tracing::debug!("Couldn't read package file. It might intentionally not exist.");
-                // Packages don't have to use JS.
-                let fns = v8::Array::new(scope, 3);
-                let undefined = v8::undefined(scope).into();
-                for fn_idx in 0..3 {
-                    fns.set_index(scope, fn_idx, undefined).ok_or_else(|| {
-                        Error::PackageImport(
-                            path.clone(),
-                            format!(
-                                "Could not set Undefined value at index {fn_idx} on package \
-                                 function array"
-                            ),
-                        )
-                    })?;
-                }
-
-                return Ok(JsPackage { fns });
-            }
-        };
-
-        // Avoid JS ReferenceError by wrapping potentially undeclared variables with `typeof`.
-        // Double braces like `{{` are Rust's escape string for a single literal `{`.
-        let wrapped_code = new_js_string(
-            scope,
-            &format!(
-                "((hash_util, hash_stdlib) => {{{code}
-                    return [
-                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
-                        typeof start_sim === 'undefined' ? undefined : start_sim,
-                        typeof run_task === 'undefined' ? undefined : run_task
-                    ]
-                }})",
-            ),
-        );
 
-        let pkg: Function<'_> = {
+        let namespace: Object<'_> = {
             let mut try_catch = v8::TryCatch::new(scope);
+            let module_map = try_catch
+                .get_slot::<Rc<RefCell<ModuleMap>>>()
+                .expect("ModuleMap is not present in isolate slots")
+                .clone();
+            let pkg = match ModuleMap::import_module(module_map, &mut try_catch, &path)? {
```

Comment:
We are calling this function always by getting the module map before, can't we move that code inside the function?

Reply:
Refactored in https://github.com/hashintel/hash/pull/493/commits/21ddfe964dccab4810645c2cb624d6d29dd03bd9